### DISCUSSION TITLE
Issue with Address Lookup Discrepancy Between Local Server and Nominatim Search

### DISCUSSION THREAD

**Discussion by rsong0606** [osm-search/Nominatim#3462] [General]
Description
I followed the [Nominatim import instructions](https://nominatim.org/release-docs/develop/admin/Import/) to import databases for different regions, specifically the US and UK, into our server. However, I noticed that some addresses cannot be found on our server even though the database has been imported successfully. Interestingly, these same addresses can be found using the Nominatim search.

Steps to Reproduce:
**Download the latest OSM data file for a region (e.g., France):**
`wget https://download.geofabrik.de/europe/france-latest.osm.pbf`
**Navigate to the Nominatim build location where the data file resides.
Run the data file with the following command:**
`nominatim add-data --osm2pgsql-cache 18000 --file luxembourg-latest.osm.pbf 2>&1 | tee luxembourg.log`
**Observations**
I tested several addresses for this region by searching on both the Nominatim search and our hosted server. Approximately 50% of the addresses return results on Nominatim search but not on our server.

**Expected Outcome**
The hosted server should return address results consistent with those found on the Nominatim search.

**Actual Outcome**
There is a discrepancy where around 50% of the addresses are not found on the hosted server.

**Question**
Do we know what could be causing this performance discrepancy and how to resolve it?

**Comment by mtmail** [comment]
Which version of Nominatim did you install? Can you give examples of addresses that aren't found? Is 'not found' 0 results or do the results differ?

**Reply by rsong0606** [reply]
Thanks for replying. The Nominatim version is 4.3.2-0.

The definition of addresses aren't found will be either nothing returns from Nominatim or null from my hosted server. I will make sure to provide an example for illustration. Additionally, I am using a structured search for both Nominatim and the hosted server. 


It is an address verification service that takes 2 addresses for comparison. Final output will be either match or mismatch. 

Nominatim search:
https://nominatim.openstreetmap.org/ui/search.html?street=1+Central+Road&city=Plymouth&country=United+Kingdom&postalcode=pl1+3bx


![WeChate134ac01b8cd612ff272eaafd8f21330](https://github.com/osm-search/Nominatim/assets/58720071/68146471-ad88-4631-a98f-4468ffac4a41)
```
{
    "results": [
        {
            "id": 0,
            "comparison_output": {
                "street_line1_match": true,
                "city_match": true,
                "state_match": true,
                "country_match": true,
                "postal_code_match": true
            },
            "address_one_id": 1,
            "address_two_id": 1,
            "address_1": "1 Central Road,Plymouth,United Kingdom,PL1 3BX",
            "address_2": "one central road,plymouth,united kingdom,pl1 3bx"
        }
    ]
}
```

Hosted server result(s):

```
{
    "results": [
        {
            "id": 0,
            "comparison_output": {
                "street_line1_match": true,
                "city_match": null,
                "state_match": null,
                "country_match": null,
                "postal_code_match": null
            },
            "address_one_id": 1,
            "address_two_id": 1,
            "address_1": "1 Central Road,Plymouth,United Kingdom,PL1 3BX",
            "address_2": "one central road,plymouth,united kingdom,pl1 3bx"
        }
    ]
}
```

There are more cases like above, which works in Nominatim only. Do you think we need more than uploading the required DB?

Thanks

**Comment by mtmail** [comment]
OpenStreetMap data isn't complete, so true address verification will not be possible. If an address isn't in OpenStreetMap data that doesn't mean it doesn't exist (and vice versa). Nominatim also does interpolation sometimes, so it will guess where a house number is if it knows the lowest and highest house number on a street, that house number might not exist at all.

Nominatim 4.4 got better with address parsing compared to 4.3. There's at least 6 months development between the version you installed and what you see on nominatim.openstreetmap.org (actually more when comparing to the git master version).

>                "city_match": true,
>                "state_match": true,

Note that sometimes a city is also a state (Berlin, Germany or Hamburg, Germany) and not every address has a city (for example addresses on islands. In the USA it's common that villages belong to counties, not cities in rural areas).

I think the data import itself worked fine. The query parsing logic keeps changing between versions, usually it gets better.

**Reply by rsong0606** [reply]
@mtmail  Thanks for your reply. I will update the Nominatim version to 4.4 and give it a try.  Hopefully, the discrepancy issue will be solved. Will keep you posted

**Comment by lonvia** [comment]
Can you please paste the request URL and response output of the request to your server instead of the results of the comparison script?
